src/sort-scripts.ts

```diff
@@ -3,31 +3,57 @@ import orderBy from 'sort-order';
 import { PackageJson } from './types';
 
 const PRE_OR_POST_PREFIX = /^(pre|post)/;
+const NPM_RUN_ALL_SEPARATOR = /([:/])/;
+const NPM_BUILTIN_SCRIPTS = ['install', 'prepare', 'pack', 'publish', 'restart', 'start', 'stop', 'test', 'version', 'uninstall'];
+
+function parseScriptName(scripts: PackageJson['scripts'], script: string): { prefix: '' | 'pre' | 'post'; name: string } {
+  if (script === 'prepublishOnly') return { prefix: 'pre', name: 'publish' };
+  const prefixMatch = PRE_OR_POST_PREFIX.exec(script);
+  const prefix = prefixMatch ? prefixMatch[0] as 'pre' | 'post' : '';
+  const name = script.slice(prefix.length);
+  if (scripts!.hasOwnProperty(name) || NPM_BUILTIN_SCRIPTS.includes(name)) return { prefix, name };
```

Comment:
I had always assumed that the `scripts` property would be present.

The `parseScriptName` function is called from within functions that create `scriptName` and `prePostHooks`.
These functions are called using `keys.sort(order)` . If `scripts` is empty, `keys.length` would be equal to 0, meaning that the `sort` method would not be called.
This means that `scripts` will always be present when `parseScriptName` is called.

However, I understand that it may be difficult to understand this assumption, so I will fix this by using optional chaining `?.` instead.

Reply:
I cannot make this change because `al` and `bl` are also used on line 32.

However, I realize that the problem is with the short variable names that I chose. I will also fix this issue.